### ISSUE TITLE
Inconsistent Linker flags and ICache usage

### ISSUE THREAD

**Issue by ProfFan** [esp-rs/esp-hal#945]
```rust
    // ideally these should be configurable
    const CONFIG_ESP32S3_INSTRUCTION_CACHE_SIZE: u32 = 0x4000; // ESP32S3_INSTRUCTION_CACHE_16KB
    const CONFIG_ESP32S3_ICACHE_ASSOCIATED_WAYS: u8 = 8; // ESP32S3_INSTRUCTION_CACHE_8WAYS
    const CONFIG_ESP32S3_INSTRUCTION_CACHE_LINE_SIZE: u8 = 32; // ESP32S3_INSTRUCTION_CACHE_LINE_32B

    // Configure the mode of instruction cache: cache size, cache line size.
    rom_config_instruction_cache_mode(
        CONFIG_ESP32S3_INSTRUCTION_CACHE_SIZE,
        CONFIG_ESP32S3_ICACHE_ASSOCIATED_WAYS,
        CONFIG_ESP32S3_INSTRUCTION_CACHE_LINE_SIZE,
    );
```

But in `memory.x`:

```
/* reserved for ICACHE */
RESERVE_ICACHE = 0x8000;
```

**Comment by bjoernQ** [comment]
Yes, it should be configurable and the linker script should honor that configuration. Currently the linker script is configured for the "worst case"

**Comment by MabezDev** [comment]
If it's not currently configurable, should we just go ahead and align the two values? We may as well use the full 0x8000 as a cache if we're reserving it?

**Comment by bjoernQ** [comment]
I guess yes, we can change the code to configure the whole reserved area as cache. However, that is only done when using PSRAM currently. Probably we should configure the cache for all chips always before running the user code

**Comment by bjoernQ** [comment]
> Yes, it should be configurable and the linker script should honor that configuration. Currently the linker script is configured for the "worst case"

I also think configuring the linker script for the "worst case" is what we need until it's configurable since we don't know what settings the 2nd stage bootloader is using (even we can know how the default bootloaders in espflash are built - users can bring their own)

**Comment by jessebraham** [comment]
Closed by #1169 (I think)
